templates/problem/voting-controls.html

```diff
@@ -0,0 +1,70 @@
+<div class="vote-form" id="id_vote_form_box">
+    <div>
+        <div id="id_has_voted_prompt">
+            <span class="vote-form-title">{{ _('Change Vote') }}</span>
+            <div id="id_date_box">
+                <span id="id_vote_date"></span>
+            </div>
+        </div>
+        <div id="id_has_not_voted_prompt">
+            <span class="vote-form-title">{{ _('Cast Vote') }}</span>
+        </div>
+        <hr style="margin-bottom: 0px !important;">
+    </div>
+    <form id="id_delete_vote_form">
+        {% csrf_token %}
+    </form>
+    <br>
+    <form id="id_vote_form">
+        {% csrf_token %}
+        <span class="vote-form-info">{{ _('Points:') }}</span>
+        <input id="id_points" class="vote-form-input-text" type="number" step="1"
+               min="{{ min_possible_vote }}" max="{{ max_possible_vote }}" name="points">
+        <br>
+        <div id="id_points_error_box">
+            <br>
+            <span id="id_points_error" class="voting-form-error"></span>
+            <br>
+        </div>
+        <br>
+        <span class="vote-form-info">{{ _('Note:') }}</span>
+        <br>
+        <br>
+        <textarea id="id_note" class="vote-form-input-text" name="note" rows="6" cols="80"></textarea>
+        <br>
+        <div id="id_note_error_box">
+            <br>
+            <span id="id_note_error" class="voting-form-error"></span>
+            <br>
+        </div>
+        <br>
+        <div>
+            <input type="submit" id="id_vote_form_submit_button" class="vote-button" value="{{ _('Vote') }}">
+            <input type="submit" id="id_delete_vote_form_submit_button" value="{{ _('Delete vote') }}" form="id_delete_vote_form">
+        </div>
+    </form>
+</div>
+
+<div>
+    <a class="form-button" id="id_vote_button"></a>
+</div>
+
+{% if vote is not none %}
+    {{ vote.note|json_script('voted_note') }}
+    <script type="text/javascript">
+        set_default_vote_data({{ vote.points }}, new Date('{{ vote.vote_time.isoformat() }}'));
+    </script>
+{% endif %}
+
+<script type="text/javascript">
+    let translated_strings = {
+        'change_vote': '{{ _('Change vote') }}',
+        'vote_points': '{{ _('Vote on problem points') }}',
+        'note_justification_help_text': "{{ _("A short justification for your vote to this problem's point value (Optional).") }}",
+        'last_voted': '{{ _('Last voted on') }}',
+    }
+    set_translated_strings(translated_strings);
```

Comment:
Sorry for this, but why don't we use [Django's Javascript translation features](https://docs.djangoproject.com/en/2.2/topics/i18n/translation/#internationalization-in-javascript-code).

Reply:
I agree, we should be using Django's built-in system instead of passing the strings in the template.

Reply:
fixed, and `set_translated_strings` is removed

Reply:
fixed, and `set_translated_strings` is removed